### PR TITLE
Adds a rest call for enabling/disabling endpoint features.

### PR DESCRIPTION
**Pull request by gpolitis**
Depends on https://github.com/jitsi/jitsi-media-transform/pull/169

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/AbstractEndpoint.java [hunk 1]
```diff
@@ -409,4 +410,6 @@ public JSONObject getDebugState()
 
         return debugState;
     }
+
+    public abstract void setFeature(EndpointDebugFeatures feature, boolean b);
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/AbstractEndpoint.java]: `b` -> `enabled`

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/EndpointDebugFeatures.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     * this enum from its value.  This assumes that the String values are derived using
+     * the following transformation of the 'keys':
+     * 1) lower case
+     * 2) underscores are replaced with hyphens
+     *
+     * @param value the String value of the enum
+     * @return an instance of the enum, if one can be derived by reversing the transformation
+     * detailed above
+     */
+    public static DebugFeatures fromString(String value)
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/EndpointDebugFeatures.java]: This is still referring to `DebugFeatures`, I'm surprised it was working?

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java [hunk 3]
```diff
@@ -63,6 +65,32 @@ public Response enableFeature(@PathParam("feature") DebugFeatures feature)
         return Response.ok().build();
     }
 
+    @POST
+    @Path("/{confId}/{epId}/enable/{feature}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response enableEndpointFeature(
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: We'll want one to disable as well I think, yeah?

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/EndpointDebugFeatures.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package org.jitsi.videobridge.rest.root.colibri.debug;
```
Comment (bgrozev) [inline src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/EndpointDebugFeatures.java]: License header

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java (hunk 3 above)
Comment (gpolitis) [reply src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: yes! shall i split the method or this approach is fine?

Comment (bbaldino) [reply src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: I think this is fine.

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java [hunk 5]
```diff
@@ -66,12 +66,13 @@ public Response enableFeature(@PathParam("feature") DebugFeatures feature)
     }
 
     @POST
-    @Path("/{confId}/{epId}/enable/{feature}")
+    @Path("/{confId}/{epId}/{state}/{feature}")
     @Consumes(MediaType.APPLICATION_JSON)
     public Response enableEndpointFeature(
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: Can you add some unit tests to `DebugTest` for this new endpoint?

Comment (gpolitis) [reply src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: does any DebugTest run on your machine? Here Mockito is failing with:

```

org.mockito.exceptions.base.MockitoException: 
Mockito cannot mock this class: class org.jitsi.videobridge.util.VideobridgeProvider.

Mockito can only mock non-private & non-final classes.
If you're not sure why you're getting this error, please report to the mailing list.


Java               : 11
JVM vendor name    : AdoptOpenJDK
JVM vendor version : 11.0.5+10
JVM name           : OpenJDK 64-Bit Server VM
JVM version        : 11.0.5+10
JVM info           : mixed mode
OS name            : Mac OS X
OS version         : 10.15.2


Underlying exception : java.lang.UnsupportedOperationException: Cannot define class using reflection

	at org.jitsi.videobridge.rest.root.colibri.debug.DebugTest.configure(DebugTest.java:43)
	at org.glassfish.jersey.test.JerseyTest.configureDeployment(JerseyTest.java:420)
	at org.glassfish.jersey.test.JerseyTest.<init>(JerseyTest.java:183)
	at org.jitsi.videobridge.rest.root.colibri.debug.DebugTest.<init>(DebugTest.java:34)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:217)
	at org.junit.runners.BlockJUnit4ClassRunner$1.runReflectiveCall(BlockJUnit4ClassRunner.java:266)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: java.lang.UnsupportedOperationException: Cannot define class using reflection
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$Unavailable.defineClass(ClassInjector.java:821)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.inject(ClassInjector.java:185)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:187)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:79)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:4376)
	at org.mockito.internal.creation.bytebuddy.SubclassBytecodeGenerator.mockClass(SubclassBytecodeGenerator.java:94)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:37)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator$1.call(TypeCachingBytecodeGenerator.java:34)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:138)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:346)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:161)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:355)
	at org.mockito.internal.creation.bytebuddy.TypeCachingBytecodeGenerator.mockClass(TypeCachingBytecodeGenerator.java:32)
	at org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker.createMockType(SubclassByteBuddyMockMaker.java:71)
	at org.mockito.internal.creation.bytebuddy.SubclassByteBuddyMockMaker.createMock(SubclassByteBuddyMockMaker.java:42)
	at org.mockito.internal.creation.bytebuddy.ByteBuddyMockMaker.createMock(ByteBuddyMockMaker.java:25)
	at org.mockito.internal.util.MockUtil.createMock(MockUtil.java:35)
	at org.mockito.internal.MockitoCore.mock(MockitoCore.java:63)
	at org.mockito.Mockito.mock(Mockito.java:1729)
	at org.mockito.Mockito.mock(Mockito.java:1642)
	... 25 more
Caused by: java.lang.IllegalArgumentException: Unknown Java version: 11
	at net.bytebuddy.ClassFileVersion.ofJavaVersion(ClassFileVersion.java:135)
	at net.bytebuddy.ClassFileVersion$VersionLocator$ForJava9CapableVm.locate(ClassFileVersion.java:337)
	at net.bytebuddy.ClassFileVersion.ofThisVm(ClassFileVersion.java:147)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$CreationAction.run(ClassInjector.java:301)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$CreationAction.run(ClassInjector.java:290)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.<clinit>(ClassInjector.java:70)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:184)
	... 42 more

```

Comment (bbaldino) [reply src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: Master runs fine for me, but it looks like you're using java 11? That's likely the problem.

File: src/test/java/org/jitsi/videobridge/rest/root/colibri/debug/DebugTest.java [hunk 6]
```diff
@@ -75,7 +89,16 @@ public void testEnableNonexistentDebugFeature()
     @Test
     public void testDisableDebugFeature()
     {
-        Response resp = target(BASE_URL + "/disable/" + DebugFeatures.PAYLOAD_VERIFICATION.getValue())
+        Response resp = target(BASE_URL + "/foo/bar/disable/" + DebugFeatures.PAYLOAD_VERIFICATION.getValue())
```
Comment (bbaldino) [inline src/test/java/org/jitsi/videobridge/rest/root/colibri/debug/DebugTest.java]: Can you add some negative test cases as well? E.g.: one where the state isn't valid and one where the feature name isn't valid.

File: src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java (hunk 5 above)
Comment (JonathanLennox) [reply src/main/java/org/jitsi/videobridge/rest/root/colibri/debug/Debug.java]: For Java 11 we'll need to update to newer versions of mockito / easymock / powermock, but everything works fine once we do so.

File: src/test/java/org/jitsi/videobridge/rest/root/colibri/debug/DebugTest.java (hunk 6 above)
Comment (gpolitis) [reply src/test/java/org/jitsi/videobridge/rest/root/colibri/debug/DebugTest.java]: done

**Comment by gpolitis** [comment]
I had to update jmt, @bbaldino could you please re-approve?

**Pull request merged**